modules/ethereum/src/mock.rs

```diff
@@ -0,0 +1,168 @@
+}
+
+parameter_types! {
+	pub const TestAuraConfiguration: AuraConfiguration = test_aura_config();
+	pub const TestValidatorsConfiguration: ValidatorsConfiguration = test_validators_config();
+}
+
+impl Trait for TestRuntime {
+	type AuraConfiguration = TestAuraConfiguration;
+	type ValidatorsConfiguration = TestValidatorsConfiguration;
+	type OnHeadersSubmitted = ();
+}
+
+/// Aura configuration that is used in tests by default.
+pub fn test_aura_config() -> AuraConfiguration {
+	AuraConfiguration {
+		empty_steps_transition: u64::max_value(),
+		strict_empty_steps_transition: 0,
+		validate_step_transition: 0x16e360,
+		validate_score_transition: 0x41a3c4,
+		two_thirds_majority_transition: u64::max_value(),
+		min_gas_limit: 0x1388.into(),
+		max_gas_limit: U256::max_value(),
+		maximum_extra_data_size: 0x20,
+	}
+}
+
+/// Validators configuration that is used in tests by default.
+pub fn test_validators_config() -> ValidatorsConfiguration {
+	ValidatorsConfiguration::Single(ValidatorsSource::List(validators_addresses(3)))
+}
+
+/// Genesis header that is used in tests by default.
+pub fn genesis() -> Header {
+	Header {
+		seal: vec![vec![42].into(), vec![].into()],
+		..Default::default()
+	}
+}
+
+/// Build default i-th block, using data from runtime storage.
+pub fn block_i(number: u64, validators: &[KeyPair]) -> Header {
+	custom_block_i(number, validators, |_| {})
+}
+
+/// Build custom i-th block, using data from runtime storage.
+pub fn custom_block_i(number: u64, validators: &[KeyPair], customize: impl FnOnce(&mut Header)) -> Header {
+	let validator_index: u8 = (number % (validators.len() as u64)) as _;
+	let mut header = Header {
+		number,
+		parent_hash: HeadersByNumber::get(number - 1).unwrap()[0].clone(),
+		gas_limit: 0x2000.into(),
+		author: validator(validator_index).address().to_fixed_bytes().into(),
+		seal: vec![vec![number as u8 + 42].into(), vec![].into()],
+		difficulty: number.into(),
+		..Default::default()
+	};
+	customize(&mut header);
+	signed_header(validators, header, number + 42)
+}
+
+/// Build signed header from given header.
+pub fn signed_header(validators: &[KeyPair], mut header: Header, step: u64) -> Header {
+	let message = header.seal_hash(false).unwrap();
+	let validator_index = (step % validators.len() as u64) as usize;
+	let signature = sign(validators[validator_index].secret(), &message.as_fixed_bytes().into()).unwrap();
+	let signature: [u8; 65] = signature.into();
+	let signature = H520::from(signature);
+	header.seal[1] = rlp_encode(&signature);
+	header
+}
+
+/// Return key pair of given test validator.
+pub fn validator(index: u8) -> KeyPair {
+	KeyPair::from_secret(Secret::from([index + 1; 32])).unwrap()
+}
+
+/// Return key pairs of all test validators.
+pub fn validators(count: u8) -> Vec<KeyPair> {
+	(0..count as usize).map(|index| validator(index as u8)).collect()
+}
+
+/// Return addresses of all test validators.
+pub fn validators_addresses(count: u8) -> Vec<Address> {
```

Comment:
```suggestion
pub fn validator_addresses(count: u8) -> Vec<Address> {
```

double plural sounds okay for me (:slavic:), but afair it's not how you do it in english though :man_shrugging: